core/include/scipp/core/element/to_unit.h

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#pragma once
+
+#include "scipp/common/overloaded.h"
+#include "scipp/core/element/arg_list.h"
+#include "scipp/core/time_point.h"
+#include "scipp/core/transform_common.h"
+
+namespace scipp::core::element {
+
+namespace {
+template <class T>
+constexpr auto round = [](const auto x) {
+  if constexpr (std::is_integral_v<T>)
```

Comment:
The branch is the wrong way around!

Reply:
Why?

Reply:
If it is an integral, you add a float and round, it it is a float, you truncate.

Reply:
Oh, no! I mixed up `T` and `decltype(x)`!

Reply:
Ok, I misunderstood what the if is doing. I confused `T` and `decltype(x)`